patches/server/0807-Add-PlayerShieldDisableEvent.patch

```diff
@@ -0,0 +1,75 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cryptite <cryptite@gmail.com>
+Date: Mon, 20 Sep 2021 13:27:50 -0500
+Subject: [PATCH] Add PlayerShieldDisableEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/Mob.java b/src/main/java/net/minecraft/world/entity/Mob.java
+index 8a864238e154e2131834d013652746b7e7a78c97..91afdf9b0bfd44e2dbfd1c36cbd35b50a6fbb407 100644
+--- a/src/main/java/net/minecraft/world/entity/Mob.java
++++ b/src/main/java/net/minecraft/world/entity/Mob.java
+@@ -1588,9 +1588,16 @@ public abstract class Mob extends LivingEntity {
+             float f = 0.25F + (float) EnchantmentHelper.getBlockEfficiency(this) * 0.05F;
+ 
+             if (this.random.nextFloat() < f) {
++                // Paper start
++                io.papermc.paper.event.player.PlayerShieldDisableEvent event = new io.papermc.paper.event.player.PlayerShieldDisableEvent((org.bukkit.entity.Player) player.getBukkitEntity(), getBukkitEntity());
++                event.callEvent();
++
++                if (!event.isCancelled()) {
```

Comment:
We could just `if(event.isCancelled()) return;` here, can't we ? there is nothing left happening in this method.

Reply:
Wasn't sure Paper's stance on early outs like that. I'm certainly fine with it.

Reply:
I guess neither am I, I guess wait until we have a maintainer/developer comment on this.

Reply:
This should be combined with the previous line. If (event.callEvent) {...